cocotb/handle.py

```diff
@@ -734,19 +735,17 @@ def _set_value(self, value, call_sim):
 
         if isinstance(value, int):
             min_val, max_val = _value_limits(len(self), _Limits.VECTOR_NBIT)
-            if min_val <= value <= max_val:
-                if len(self) <= 32:
-                    call_sim(self, self._handle.set_signal_val_int, set_action, value)
-                    return
-
-                if value < 0:
-                    value = BinaryValue(value=value, n_bits=len(self), bigEndian=False, binaryRepresentation=BinaryRepresentation.TWOS_COMPLEMENT)
-                else:
-                    value = BinaryValue(value=value, n_bits=len(self), bigEndian=False, binaryRepresentation=BinaryRepresentation.UNSIGNED)
+            if value < min_val or max_val < value:
```

Comment:
```suggestion
            if value < min_val or value > max_val:
```

would be sightly easier to read IMO.